quill-core/src/main/scala/io/getquill/norm/BetaReduction.scala

```diff
@@ -19,6 +19,14 @@ case class BetaReduction(map: collection.Map[Ast, Ast])
           case _ => apply(values(name.drop(1).toInt - 1))
         }
 
+      case Property(CaseClass(tuples), name) =>
+        val aliases = tuples.map(_._2).distinct
+        aliases match {
+          case alias :: Nil if tuples.size > 1 =>
```

Comment:
could you give some context on why this special case is necessary?

Reply:
I don't think it's necessary actually, I got it from the Tuple handling code. As far as I understand it is an optimization in the case that all the elements of the Case Class are the same thing, the beta reduction only has to recurse once (i.e. if all the elements of the AST are the same thing). As far as I understand it's not strictly necessary. Would you like me to take it out?

Reply:
It seems that it was introduced by https://github.com/getquill/quill/pull/149. I don't understand why it's necessary for tuples, I think we could not have it for case classes unless we discover the reason.

Reply:
I guess I'll just take it out for now. What about the BetaReduction unit test "deduplicates aliases in secondary table join" though? Generally speaking, if a robust de-duping of aliases is desired in this part of the reduction, why not put all the aliases into a Set and then recurse over the distinct values? Right now it's a bi-modal check whether either all values are identical and de-duping is done or there's at least one non-identical value and no de-duping is done.

Reply:
I don't understand this code and test. We don't need to worry about alias conflict in beta reduction since it factors the scope of values (including shadowing).